folding-schemes/examples/external_inputs.rs

```diff
@@ -0,0 +1,213 @@
+#![allow(non_snake_case)]
```

Comment:
Just curious about the use case, this example could also be rewritten in the way the circuit handles the hash chain of signatures itself without an external input right?

Reply:
This is kind of a simple example using only hashes, but it showcases the approach that could be used for example to verify multiple signatures, one for each folding step. The FCircuit would check the signature and is folded for each of the signatures and public keys.
But: the signature is inputted from the external inputs, which are not part of the initial state.
So, following the [diagram from line 45](https://github.com/privacy-scaling-explorations/folding-schemes/pull/78/files#diff-0d0653ec716e47765490702600988ef72456e5a687824e0534afa0be0b8dd92dR45), the `w_i` "are not part" of the state, although internally they are used "as part" of the state to compute the new state.

The main difference in this PR is that now the FCircuit has access to know the `i` at each step internally (both at `step_native` and `generate_step_constraints`), which allows it to get specific values from some array, in the case of this example the circuit has this `external_inputs` array, which internally in the circuit are accessed at the `i` position.

I'm not sure if this is what you were referring to, does it make sense?

Reply:
I see, thanks for explanation!